protocols/pop3/src/main/java/org/apache/james/protocols/pop3/core/POP3MessageCommandArguments.java

```diff
@@ -0,0 +1,77 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.protocols.pop3.core;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.james.protocols.api.Request;
+
+public class POP3MessageCommandArguments {
+    
+    private static POP3MessageCommandArguments INVALID = new POP3MessageCommandArguments(false, 0, Optional.empty());
+    
+    private final boolean valid;
+    private final int messageNumber;
+    private final Optional<Integer> lineCount;
+
+    public static POP3MessageCommandArguments fromRequest(Request request) {
+        String parameters = request.getArgument();
+        if (parameters == null) {
+            return INVALID;
+        }
+
+        try {
+            List<Integer> args = Arrays.stream(parameters.trim().split(" "))
+                .map(String::trim).filter(arg -> !arg.isEmpty()).map(Integer::parseInt)
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
```suggestion
            List<Integer> args = Splitter.on(' ')
            .omitEmptyStrings()
            .trimResults()
            .splitToList(parameters)
            .stream()
            .map(Integer::parseInt)
                .collect(ImmutableList.toImmutableList());
```

Rationnals:
 - `String::split` hides a regex compilation and evaluation that I would better avoid on each POP3 command
 - Splitter is a common, readable pattern we use in many other places
 - Unmodifiable list is a less strong version of immutable list, we should rather use the later.
 
 (Sorry for being picky)

Reply:
Now that you mention it, I remember we talked about this before. Committed.

Reply:
... needs imports as well. Fixed in latest push.